src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/CodeGeneration/CodeWriter.IndentCache.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Language.CodeGeneration;
+
+public sealed partial class CodeWriter
+{
+    private static class IndentCache
+    {
+        private const int MaxCachedIndentSize = 80;
+
+        // Caches for combination of useTabs and common tab sizes (2, 4)
+        private static ReadOnlyMemory<char>?[]? s_tabsTwo;
+        private static ReadOnlyMemory<char>?[]? s_tabsFour;
+        private static ReadOnlyMemory<char>?[]? s_spacesTwo;
+        private static ReadOnlyMemory<char>?[]? s_spacesFour;
```

Comment:
It feels a bit overkill to make these arrays of nullable `ReadOnlyMemory<char>?`. We shouldn't allow zero-length indents, so it should be fine to make this non-nullable `ReadOnlyMemory<char>[]`.